raphtory/src/algorithms/components/in_components.rs

```diff
@@ -91,13 +96,85 @@ where
     );
     AlgorithmResult::new(graph.clone(), "In Components", results_type, res)
 }
+/// Computes the in-component of a given node in the graph
+///
+/// # Arguments
+///
+/// * `node` - The node whose in-component we wish to calculate
+///
+/// Returns:
+///
+/// A Vec containing the Nodes within the given nodes in-component
+///
+pub fn in_component<'graph, G: GraphViewOps<'graph>>(node: NodeView<G>) -> Vec<NodeView<G>> {
+    let mut in_components = HashSet::new();
+    let mut to_check_stack = Vec::new();
+    node.in_neighbours().iter().for_each(|node| {
+        let id = node.node;
+        in_components.insert(id);
```

Comment:
This could be made much more efficient by actually using the `bool` return from the insert. We only need to push the node to the `to_check_stack` when it wasn't already in the set.